core/execution/common/market_activity_tracker.go

```diff
@@ -40,27 +40,54 @@ type marketTracker struct {
 	readyToDelete          bool
 }
 
-// MarketActivityTracker tracks how much fees are paid and received for a market by parties by epoch.
+// MarketActivityTracker tracks how much fees are paid and received for a market
+// by parties by epoch.
 type MarketActivityTracker struct {
 	log                *logging.Logger
 	marketToTracker    map[string]*marketTracker
 	eligibilityChecker EligibilityChecker
 	currentEpoch       uint64
 	ss                 *snapshotState
+
+	// notionalTakerVolumeByParty tracks the notional taker volume by party for
+	// the current epoch. It's reset at every epoch start.
+	notionalTakerVolumeByParty map[types.PartyID]*num.Uint
 }
 
-// NewFeesTracker instantiates the fees tracker.
 func NewMarketActivityTracker(log *logging.Logger, epochEngine EpochEngine) *MarketActivityTracker {
 	mat := &MarketActivityTracker{
 		marketToTracker: map[string]*marketTracker{},
 		ss:              &snapshotState{},
 		log:             log,
 	}
+
+	mat.clearNotionalTakerVolume()
+
 	epochEngine.NotifyOnEpoch(mat.onEpochEvent, mat.onEpochRestore)
 	return mat
 }
 
-// GetProposer returns the proposer of the market or empty string if the market doesn't exist.
+func (mat *MarketActivityTracker) TrackNotionalTakerVolume(party types.PartyID, volumeToAdd *num.Uint) {
```

Comment:
this should definitely not be merged. Can you not just use a dummy tracker and pass it to there referral engine? 

why?
1. the market_activity_tracker is radically changing in the reward PR - I can add it in there properly
2. it's missing here snapshot support
3. it's missing here the integration to the market which feeds it

Reply:
Noted ! Will use a dummy.